Week4/MyFavoriteMovies/app/src/main/java/com/raywenderlich/myfavoritemovies/MovieDetailAdapter.kt

```diff
@@ -0,0 +1,24 @@
+package com.raywenderlich.myfavoritemovies
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+
+class MovieDetailAdapter(var movie: Movie?) : RecyclerView.Adapter<MovieDetailViewHolder>() {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieDetailViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.movie_detail_view_holder, parent, false)
+        return MovieDetailViewHolder(view)
+    }
+
+    override fun getItemCount(): Int {
+        return 1
+    }
+
+    override fun onBindViewHolder(holder: MovieDetailViewHolder, position: Int) {
+        movie?.poster?.let { holder.moviePosterImageView.setImageResource(it) }
```

Comment:
Avoid using `?.let`, it's generally an antipattern. I know some videos might've used it, and you can find examples online, but it's much better having a simple nullcheck instead!